pyproject.toml

```
@@ -17,7 +17,7 @@ repository = "https://github.com/MeltanoLabs/meltano-map-transform"
 documentation = "https://github.com/MeltanoLabs/meltano-map-transform#readme"
 
 [tool.poetry.dependencies]
-python = "<3.11,>=3.7.1"
+python = "<3.12,>=3.7.1"
```

What if we start practicing only a lower bounds here?

```suggestion
python = ">=3.7.1"
```

And add the list of 'tested' python versions to package metadata?

Wdyt?

---

@aaronsteers I tried it but `poetry lock` doesn't like the different constraints:

```
The current project's Python requirement (>=3.7.1) is not compatible with some of the required packages Python requirement:
  - singer-sdk requires Python >=3.7.1,<3.12, so it will not be satisfied for Python >=3.12

Because meltano-map-transform depends on singer-sdk (0.19.0) which requires Python >=3.7.1,<3.12, version solving failed.

  • Check your dependencies Python requirement: The Python requirement can be specified via the `python` or `markers` properties

    For singer-sdk, a possible solution would be to set the `python` property to ">=3.7.1,<3.12"

    https://python-poetry.org/docs/dependency-specification/#python-restricted-dependencies,
    https://python-poetry.org/docs/dependency-specification/#using-environment-markers
```